web/src/stream_ui_updates.js

```diff
@@ -155,6 +156,25 @@ export function enable_or_disable_permission_settings_in_edit_panel(sub) {
     );
 }
 
+export function update_announce_stream_option() {
+    if (!hash_util.is_create_new_stream_narrow()) {
+        return;
+    }
+    if (stream_data.get_notifications_stream() === "") {
+        $("#announce-new-stream .checkbox").hide();
```

Comment:
I think we can hide and show the `$("#announce-new-stream")` element directly. Otherwise, the live-update does not work correctly always as `$("#announce-new-stream")` has `display: none` due to the code [here](https://github.com/zulip/zulip/blob/main/web/src/stream_create.js#L127).